### ISSUE TITLE
Using put on an empty collection

### ISSUE THREAD

**Issue by Helfull** [ecrmnn/collect.js#262]
I have the following code written in typescript

```
function loadTest() {
  const collection = collect();

  collection.put(1, 'test');

  console.log(collection.count(), {collection});
}
```

now the output of this is

![image](https://user-images.githubusercontent.com/720645/96031382-8f11fc00-0e5d-11eb-972b-17bb3a943917.png)

i also tried it with a test and it failed

```
  it('should set the given key and value in an empty collection', () => {
    const collection = collect();
    const modified = collection.put('club', 'Liverpool FC');

    expect(collection).to.eql(modified);
    expect(collection.all()).to.eql({
      name: 'Roberto Firmino',
      club: 'Liverpool FC',
    });
  });
```

Question: should the collection check if the items are an array or an object?
Or should i always initialize the collection with either of them?

**Comment by Helfull** [comment]
I discovered this problem while working with typescript + vuejs + collectjs.

In typescript i cant do `new Collection({})`, as there is no typescript definition with constructor arguments.

**Comment by ecrmnn** [comment]
Yeah I've seen this too when using something like `put` on an empty collection or a collection based on an array.

Can you init your collection with `collect({})`?

**Comment by Helfull** [comment]
yeah i can init with the `collect` function but not with the constructor as there is a missing definition in the typescript definition.

**Comment by ecrmnn** [comment]
Closed in #263
